wire/msgtx.go

```diff
@@ -757,18 +963,25 @@ func (msg *MsgTx) decodeWitnessValueSigning(r io.Reader, pver uint32) error {
 		return messageError("MsgTx.decodeWitness", str)
 	}
 
+	var totalScriptSize uint64
 	txIns := make([]TxIn, count)
 	msg.TxIn = make([]*TxIn, count)
 	for i := uint64(0); i < count; i++ {
+		// The pointer is set now in case a script buffer is borrowed
+		// and needs to be returned to the pool on error.
 		ti := &txIns[i]
+		msg.TxIn[i] = ti
 		err = readTxInWitnessValueSigning(r, pver, msg.Version, ti)
 		if err != nil {
+			returnMemory()
 			return err
 		}
-		msg.TxIn[i] = ti
+		totalScriptSize += uint64(len(ti.SignatureScript))
 	}
 	msg.TxOut = make([]*TxOut, 0)
 
+	writeTxOutScriptsToMsgTx(msg, totalScriptSize)
```

Comment:
I think this needs a comment to explain what's going on since this would be quite unintuitive for a reader that is not already familiar with this code.